cmake/kokkos_enable_options.cmake

```diff
@@ -61,6 +61,7 @@ KOKKOS_ENABLE_OPTION(HIP_MULTIPLE_KERNEL_INSTANTIATIONS OFF "Whether multiple ke
 
 # This option will go away eventually, but allows fallback to old implementation when needed.
 KOKKOS_ENABLE_OPTION(IMPL_DESUL_ATOMICS   ON  "Whether to use desul based atomics - option only during beta")
+KOKKOS_ENABLE_OPTION(DESUL_ATOMICS_FORCE_EXTERNAL OFF "Whether to enfore finding an external desul installation")
```

Comment:
I dislike the option name.
I just came across https://cmake.org/cmake/help/latest/variable/CMAKE_REQUIRE_FIND_PACKAGE_PackageName.html
Wouldn't you agree this would make the new option added here obsolete?

Reply:
Yes, if we can rely on CMake >=3.22 the new option makes this obsolete.
I mainly introduced it here for unit testing, though, and CMake is not that new there.

Reply:
The option works fine for me using `CMake 3.23.2` but I agree with @nmm0 that it makes sense to force the user to tell us to use an external or internal package (with the default being internal).